flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -620,10 +620,16 @@ public static String getFormatPrefix(
             Class<T> factoryClass, DynamicTableFactory.Context context) {
         final String connectorOption = context.getCatalogTable().getOptions().get(CONNECTOR.key());
         if (connectorOption == null) {
-            throw new ValidationException(
-                    String.format(
-                            "Table options do not contain an option key '%s' for discovering a connector.",
-                            CONNECTOR.key()));
+            ManagedTableFactory factory =
```

Comment:
More than having an empty connector id and hardcoding the discovery of `ManagedTableFactory`, I wonder if we can have a connector id for that, and then we inject it from `TableEnvironmentImpl` every time we parse a table which doesn't provide a `connector` value in the table options.

Reply:
I prefer the empty string to other ids because the empty string itself is a special symbol that is often mixed up with "none".

Reply:
tbh I also don't like the empty string. A `managed-table` or even `default` (which we use for planner and executor discovery) would be nicer and looks less like a bug. `default` would ensure consistency.

Reply:
You have convinced me that I think we can use `default`